src/vs/platform/files/common/files.ts

```diff
@@ -742,13 +747,22 @@ interface IBaseStat {
 	 * it is optional.
 	 */
 	etag?: string;
+
+	/**
+	 * The file is read-only
+	 *
+	 * The value may or may not be resolved as
+	 * it is optional.
+	 */
+	readonly?: boolean;
```

Comment:
`readonly` should only be optional in `IStat` but given it is included in each `IStat` it can be required on the level of `IBaseStat`, no?

Reply:
Changes pushed. Please verify I understood you correctly.

Reply:
@bpasero I am now unsure about the wisdom of making `readonly` required in IBaseStat, IBaseStatWithMetaData and IFileStatWithMetaData. It mean that if a FSP doesn't define the new `readonly` property of FileStat I default it true or false depending on whether the FSP registered as a readonly one or not. But if the FSP subsequently unregisters and reregisters with the opposite readonly mode (now supported, thanks to your #110854), already-resolved files in the FSP report the previous default value for readonly.

I enhanced my fsprovider-sample branch to allow exploration of this.